taxi/forms.py

```diff
@@ -0,0 +1,52 @@
+from django import forms
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+from django.core.exceptions import ValidationError
+
+from taxi.models import Driver, Car
+
+
+class LicenceValidateMixin:
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]  # noqa
+
+        if not (
+            len(license_number) == 8
+            and license_number[0:3].isupper()
+            and license_number[0:3].isalpha()
+            and license_number[3:].isnumeric()
+        ):
+            raise ValidationError(
+                "The driver's license must be in the format 'XXX00000' "
+                "(3 uppercase letters and 5 numbers)"

```

Comment:
Cases are grouped, so it is not clear what exactly user did wrong when the error occurs. But I guess it is not that big of a deal, so it is up to you.

Reply:
I believe that it is impossible to ungroup, because the user will be asked to correct errors sequentially. He will not be able to come to the right decision after one mistake.
This error is not difficult to fix and there is nowhere to get lost in it, so I think that it is better to group them.
your case:
ABCD12345 -> Length error -> ABCD1234 -> Letter count error -> abc12345 -> Uppercase error.
my case:
ABCD12345 -> License format error, format and qualifications specified -> ABC12345 -> Success

Reply:
I also don't like how it's grouped together
You can make 3 exceptions:
- length is not equal to 8
- the first 3 symbols are not uppercase letters
- the last 5 characters are not numbers

A person will only see one message, but they will correct it according to that message. There will be 3 steps in the worst case